script/tool/lib/src/analyze_command.dart

```diff
@@ -113,6 +117,12 @@ class AnalyzeCommand extends PackageLoopingCommand {
       return PackageResult.skip('No lib/ directory.');
     }
 
+    if (getBoolArg(_downgradeFlag)) {
+      if (!await _runPubCommand(package, 'downgrade')) {
```

Comment:
From https://dart.dev/tools/pub/cmd/pub-downgrade:
`The dart pub downgrade command creates a lockfile. If one already exists, pub ignores that file and generates a new one from scratch, using the lowest versions of all dependencies.`

Are we okay with the analyze command creating artifacts? If so, we should note it.

Reply:
I guess pub get also does a similar thing, so this should be fine.

Reply:
Yes, analyze always creates a lock file because we always have to fetch packages.